nats-base-client/jsclient.ts

```diff
@@ -252,12 +262,18 @@ export class JetStreamClientImpl extends BaseApiClient
             timer.cancel();
             timer = null;
           }
-          if (
-            isNatsError(err) &&
-            (err.code === ErrorCode.JetStream404NoMessages ||
-              err.code === ErrorCode.JetStream408RequestTimeout)
-          ) {
-            qi.stop();
+          if (isNatsError(err)) {
+            switch (err.code) {
+              case ErrorCode.JetStream404NoMessages:
+              case ErrorCode.JetStream409:
+                qi.stop();
+                return [null, null];
```

Comment:
Does the user get no feedback on what happened if there is 409 status returned? Also, why do we return in this specific scenario?

Reply:
Got it, make sense. I assumed 404 means there are no more messages.